src/sys/socket/mod.rs

```diff
@@ -567,57 +569,95 @@ impl<'a> ControlMessage<'a> {
         }
     }
 
+    /// Returns the value to put into the `cmsg_type` field of the header.
+    fn type_(&self) -> libc::c_int {
```

Comment:
I don't like trailing underscores.  How about naming this method `cmsg_type` instead?